src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -4310,6 +4309,41 @@ GenerateSyntheticShardIntervalArray(int partitionCount)
 }
 
 
+/*
+ * Determine RowModify required for given query
+ */
+RowModify
+RowModifyForQuery(Query *query)
+{
+	switch (query->commandType)
+	{
+		case CMD_SELECT:
+		{
+			return ROW_MODIFY_READONLY;
+		}
+
+		case CMD_INSERT:
+		{
+			if (query->onConflict != NULL)
+			{
+				return ROW_MODIFY_COMMUTATIVE;
+			}
```

Comment:
Minor: without an `else` and with the `FALLTHROUGH` I need to scan more lines to understand this. I'd have preferred an `else` block here.